src/snowflake/snowpark/_internal/packaging_utils.py

```diff
@@ -0,0 +1,296 @@
+    We detect these so that we can switch to Anaconda-supported versions of these packages, where possible (or warn
+    the user if it is not possible).
+
+    The detection method looks for file extensions that correspond to native code usage (Note that this method is
+    not perfect and will result in false positives/negatives).
+
+    :param target: Target folder where installed packages are present.
+    :param downloaded_packages_dict: Mapping between package and a list of corresponding folders.
+    :return: A list of unique package names, which contain native dependencies.
+    """
+
+    def invert_downloaded_package_to_entry_map(
+        downloaded_packages_dict: Dict[Requirement, List[str]]
+    ) -> Dict[str, Set[str]]:
+        record_entry_to_package_name_map: Dict[str, Set[str]] = {}
+        for requirement, record_entries in downloaded_packages_dict.items():
+            for record_entry in record_entries:
+                if record_entry not in record_entry_to_package_name_map:
+                    record_entry_to_package_name_map[record_entry] = {requirement.name}
+                else:
+                    record_entry_to_package_name_map[record_entry].add(requirement.name)
+        return record_entry_to_package_name_map
+
+    native_libraries = set()
+    native_extensions = {
+        ".pyd",
+        ".pyx",
+        ".pxd",
+        ".dll" if platform.system() == "Windows" else ".so",
+    }
+    for native_extension in native_extensions:
+        glob_output = glob.glob(
+            os.path.join(target, "**", f"*{native_extension}")
+        ) + glob.glob(os.path.join(target, f"*{native_extension}"))
+        if glob_output:
+            folder_to_package_map = invert_downloaded_package_to_entry_map(
+                downloaded_packages_dict
+            )
+            for path in glob_output:
+                relative_path = os.path.relpath(path, target)
+
+                # Fetch record entry (either base directory or a file name if base directory is target)
+                record_entry = os.path.split(relative_path)[0]
+                if record_entry == "":
+                    record_entry = relative_path
+
+                # Check which package owns this record entry
+                if record_entry in folder_to_package_map:
+                    library_set = folder_to_package_map[record_entry]
+                    for library in library_set:
+                        if library not in native_libraries:
+                            _logger.info(f"Potential native library: {library}")
+                            native_libraries.add(library)
+    return native_libraries
+
+
+def zip_directory_contents(directory_path: str, output_path: str) -> None:
+    """
+    All files/folders inside the directory path are copied over into the zip package.
+    All files/folders installed outside the directory path, which are not the directory path or the output path or
+    hidden are also copied over.
```

Comment:
It looks like by outside, you mean one level up of directory path? The current description is vague and does not convey that.